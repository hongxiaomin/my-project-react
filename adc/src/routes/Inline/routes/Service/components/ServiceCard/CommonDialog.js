import React from 'react'
import PropTypes from 'prop-types'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { Button, Dialog } from '@blueprintjs/core'
import './Dialog.scss'

export const CommonDialog = ({ title, isOpen, toggleDialog, handleSubmit }) => (<div styleName='Dialog-container'>
  <Dialog
    isOpen={isOpen}
    onClose={toggleDialog}
    styleName='dialog'
  >
    <div>
      <h6 styleName='headerTitle'><WrapperFormattedMessage id='serviceManagement.serviceCard.confirmation' /></h6>
    </div>
    <div styleName='dialogBody'>
      <br />
      <p styleName='cn'>{title}</p>
      <div styleName='btnGroup'>
        <Button styleName='btnCancel' onClick={toggleDialog}>
          <WrapperFormattedMessage id='cancel' />
        </Button>
        <Button styleName='btnEdit' onClick={handleSubmit}>
          <WrapperFormattedMessage id='Save' />
        </Button>
      </div>
    </div>
  </Dialog>
</div>)

CommonDialog.propTypes = {
  title: PropTypes.string.isRequired,
  isOpen: PropTypes.bool.isRequired,
  toggleDialog: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
}

export default CommonDialog
