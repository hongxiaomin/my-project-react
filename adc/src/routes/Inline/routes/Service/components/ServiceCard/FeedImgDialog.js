import React from 'react'
import { connect } from 'react-redux'
import { Button, Dialog, Checkbox } from '@blueprintjs/core'
import { simpleFetch } from 'helpers/fetchWrapper'
import classes from './Dialog.scss'

type Props = {
  id: Number,
  kernelId: Number,
  isOpen: boolean,
  toggleDialog: Function,
  token: String
};

export class FeedImgDialog extends React.Component {
  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
    this.handleEnabledChange = this.handleEnabledChange.bind(this)
    this.state = {
      isContinuous: false,
      pic: '',
    }
  }

  handleSubmit() {
    let tmp = []
    if (this.state.pic === 'CSOT图片') {
      tmp = ['1.jpg', '2.jpg', '3.jpg', '4.jpg', '5.jpg', '6.jpg', '7.jpg', '8.jpg', '9.jpg', '10.jpg']
    } else {
      tmp = [
        '1F950_T5A471B012B_TCAOH110_68_-624.127_824.558_O_M_20170612_0457001.jpg',
        '1F950_T5A471B012B_TCAOH110_68_-624.127_824.558_O_M_20170612_0457002.jpg',
        '1F950_T5A471B012B_TCAOH110_68_-624.127_824.558_O_M_20170612_0457003.jpg',
        '1F950_T5A471B012B_TCAOH110_68_-624.127_824.558_O_M_20170612_0457004.jpg',
        '1F950_T5A471B012B_TCAOH110_68_-624.127_824.558_O_M_20170612_0457005.jpg',
        '1F950_T5A471B012B_TCAOH110_68_-624.127_824.558_O_M_20170612_0457006.jpg',
        '1F950_T5A471B012B_TCAOH110_68_-624.127_824.558_O_M_20170612_0457007.jpg',
        '1F950_T5A471B012B_TCAOH110_68_-624.127_824.558_O_M_20170612_0457008.jpg',
        '1F950_T5A471B012B_TCAOH110_68_-624.127_824.558_O_M_20170612_0457009.jpg',
        '1F950_T5A471B012B_TCAOH110_68_-624.127_824.558_O_M_20170612_0457010.jpg',
        '1F950_T5A471B012B_TCAOH110_68_-624.127_824.558_O_M_20170612_0457011.jpg',
        '1F950_T5A471B012B_TCAOH110_68_-624.127_824.558_O_M_20170612_0457012.jpg',
        '1F950_T5A471B012B_TCAOH110_68_-624.127_824.558_O_M_20170612_0457013.jpg',
        '1F950_T5A471B012B_TCAOH110_68_-624.127_824.558_O_M_20170612_0457014.jpg',
        '1F950_T5A471B012B_TCAOH110_68_-624.127_824.558_O_M_20170612_0457015.jpg',
      ]
    }
    const url = 'ojs/kernel/action'
    simpleFetch(url, {
      method: 'POST',
      body: {
        kernelId: this.props.kernelId.toString(),
        cmd: 'feed',
        payload: tmp,
        isContinuous: this.state.isContinuous,
      },
    })
      // .then((data) => {
      //   // this.props.dispatch(getServiceList())
      // })
    this.props.toggleDialog()
  }

  handleEnabledChange() {
    this.setState({
      isContinuous: !this.state.isContinuous,
    })
  }

  handleChange(type) {
    return (e) => {
      this.setState({ [type]: e.target.value })
    }
  }
  props: Props;

  render() {
    return (
      <div className={classes['Dialog-container']}>
        <Dialog
          isOpen={this.props.isOpen}
          onClose={this.props.toggleDialog}
          className={classes.dialog}
        >
          <div className={classes.header}>
            <h6 className={classes.headerTitle}>{'Feed Image来源'}( ID {this.props.id})</h6>
          </div>
          <div className={classes.dialogBody}>
            <br />
            <label
              className={`pt-label pt-inline ${classes.filter}`}
              style={{ paddingLeft: '80px' }}
            >
              图片类型
              <div className='pt-select'>
                {(() => (
                  <select
                    className={classes.select}
                    onChange={this.handleChange('pic')}
                  >
                    <option value=''>请选择</option>
                    <option value='CSOT图片'>CSOT图片</option>
                    <option value='焊锡检测图片'>焊锡检测图片</option>
                  </select>
                  ))()}
              </div>
            </label>
            <div className={classes.btnGroup}>
              <Checkbox
                checked={this.state.isContinuous}
                className={classes.isContinuous}
                onChange={this.handleEnabledChange}
              >
                重复循环
              </Checkbox>
              <Button className={classes.btnCancel} onClick={this.props.toggleDialog}>
                Cancel
              </Button>
              <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleSubmit}>
                Save
              </Button>
            </div>
          </div>
        </Dialog>
      </div>
    )
  }
}
export default connect(
  state => ({
    token: state.auth.token,
  }),
)(FeedImgDialog)
