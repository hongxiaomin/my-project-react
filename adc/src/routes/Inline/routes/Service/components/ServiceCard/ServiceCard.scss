.container {
  border: 1px solid #ccc;
  border-radius: 4px;
  margin: 10px 0px;
  margin-right: 10px;
  position: relative;
  
  .bg {
    position: absolute;
    width: 100%;
    height: 100%;
    left: 0;
    top: 0;
    background: rgba(0, 0, 0, .8);
    color: #fff;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 20px;
  }
}
.top {
  display: flex;
  padding: 5px 10px 0px 10px;
  justify-content: space-between;
  background-color: rgb(247, 247, 247);
}
.h5 {
  line-height: 1.5em;
  width: 65%;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}
.id {
  font-size: 16px;
  padding-left: 10px;
  // font-weight: bold;
  padding-bottom: 10px;
  background-color: rgb(247, 247, 247);
}
.progress {
  margin-bottom: 10px;
  height: 30px;
  text-align: center;
  line-height: 30px;
  color: white;
  background-color: green;
}
.detail {
  padding-bottom: 10px;
}
.textField {
  padding: 5px 20px;
  display: flex;
  // justify-content: center;
  // align-items: flex-start;
}
  .key {
    color: #909090;
    width: 40%;
    padding-right: 20px;
    // text-align: right;
  }

  .value {
    font-weight: bold;
    color: 333;
    width: 60%;
    word-break: break-all;
    overflow:hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
  }

.span {
  display: inline-block;
  padding-right: 10px;
  width: 100%;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.spin {
  animation: spin 800ms infinite linear;
}

@keyframes spin {
  0%   { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}
