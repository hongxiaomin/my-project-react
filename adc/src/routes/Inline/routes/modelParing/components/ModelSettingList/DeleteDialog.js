import React from 'react'
import { isEmpty } from 'lodash'
import { Button, Dialog } from '@blueprintjs/core'
import { connect } from 'react-redux'
import { deleteModel } from '../../modules/ModelParing'
import classes from './Dialog.scss'

type Props = {
  dataDetail: Number,
  delOpen: Boolean,
  toggleDel: Function,
  dispatch: Function
};

export class DeleteDialog extends React.Component {


  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit() {
    this.props.dispatch(deleteModel(this.props.dataDetail.original.configId))
    this.props.toggleDel()
  }
  props: Props;
  render() {
    return (
      <div className={classes['Dialog-container']}>
        {
          !isEmpty(this.props.dataDetail) &&
          <Dialog
            isOpen={this.props.delOpen}
            onClose={this.props.toggleDel}
            className={classes.dialog}
          >
            <div className={classes.header}>
              <h6 className={classes.headerTitle}>Confirmation</h6>
            </div>
            <div className={classes.dialogBody}>
              <br />
              <p className={classes.cn}>{'Are you certain that you want to delete this model pairing?'}</p>
              <div className={classes.btnGroup}>
                <Button className={classes.btnCancel} onClick={this.props.toggleDel}>
                  Cancel
                </Button>
                <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleSubmit}>
                  Save
                </Button>
              </div>
            </div>
          </Dialog>
        }
      </div>
    )
  }
}

export default connect()(DeleteDialog)
