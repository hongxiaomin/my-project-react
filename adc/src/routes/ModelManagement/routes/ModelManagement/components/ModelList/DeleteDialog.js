import React from 'react'
import { isEmpty } from 'lodash'
import { Button, Dialog } from '@blueprintjs/core'
import { connect } from 'react-redux'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { deleteModel } from '../../modules'
import classes from './Dialog.scss'

type Props = {
  dataDetail: Number,
  delOpen: Boolean,
  toggleDel: Function,
  dispatch: Function
};

export class DeleteDialog extends React.Component {


  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit() {
    this.props.dispatch(deleteModel(this.props.dataDetail.original.id))
    this.props.toggleDel()
  }
  props: Props;
  render() {
    return (
      <div className={classes['Dialog-container']}>
        {
          !isEmpty(this.props.dataDetail) &&
          <Dialog
            isOpen={this.props.delOpen}
            onClose={this.props.toggleDel}
            className={classes.dialog}
          >
            <div className={classes.header}>
              <h6 className={classes.headerTitle}><WrapperFormattedMessage id="confirmation" /></h6>
            </div>
            <div className={classes.dialogBody}>
              <br />
              <p className={classes.cn}>
                <WrapperFormattedMessage id="modelManagement.deleteDialog.note" />
              </p>
              <div className={classes.btnGroup}>
                <Button className={classes.btnCancel} onClick={this.props.toggleDel}>
                  <WrapperFormattedMessage id="cancel" />
                </Button>
                <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleSubmit}>
                  <WrapperFormattedMessage id="Save" />
                </Button>
              </div>
            </div>
          </Dialog>
        }
      </div>
    )
  }
}

export default connect()(DeleteDialog)
