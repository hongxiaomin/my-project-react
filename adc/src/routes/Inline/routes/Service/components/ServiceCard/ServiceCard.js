import React from 'react'
import PropTypes from 'prop-types'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { injectIntl } from 'react-intl'
import { connect } from 'react-redux'
import EditDialog from '../EditDialog'
import CreateServiceDialog from '../CreateServiceDialog'
import classes from './ServiceCard.scss'
import { sendCommand, resetService, deleteService } from '../../modules/Service'
import CommonDialog from './CommonDialog'

export class TextField {
  constructor(key, value) {
    this.key = key
    this.value = value
  }

  getRender(btn) {
    return (
      <div className={classes.textField}>
        <div className={classes.key}>
          {this.key}
        </div>
        <div className={classes.value}>
          {this.value}
          {btn}
        </div>
      </div>
    )
  }
}

export class ServiceCard extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      progress: '',
      status: this.props.item.status || '',
      addOpen: false,
      editOpen: false,
      deleteOpen: false,
      destroyOpen: false,
      stopOpen: false,
      startOpen: false,
      resetOpen: false,
    }
  }

  toggleDialog = type => () => {
    const key = `${type}Open`
    this.setState({
      [key]: !this.state[key],
    })
  }

  handleSubmit = type => () => {
    this.props.dispatch(sendCommand({
      serviceId: this.props.item.serviceId,
      command: type,
    }, () => {
      this.toggleDialog(type)()
    }))
  }

  handleResetSubmit = () => {
    this.props.dispatch(resetService({
      serviceId: this.props.item.serviceId,
    }))
    this.toggleDialog('reset')()
  }

  handleDeleteDialog = () => {
    this.props.dispatch(deleteService(this.props.item.deviceId))
    this.toggleDialog('delete')()
  }

  render() {
    const { item, loading, intl } = this.props
    let stopOrLoading = 'Stop'
    let status
    let color
    if (loading === 'stop') stopOrLoading = 'loading'

    switch (item.status.toLowerCase()) {
      case 'initing': case 'inited': case 'stopped': case 'stopping':
        status = 'Idle'
        color = '#ccc'
        break
      case 'starting':
        status = 'Creating'
        color = '#66cc00'
        break
      case 'started':
        status = 'Ready'
        color = '#0099ff'
        break
      case 'feeding': case 'feeded':
        status = 'In Progress'
        color = 'orange'
        break
      case '': case 'none': case 'destroyed': case 'destroying': default:
        status = 'Undefined'
        color = '#ccc'
    }

    return (
      <div style={{ width: '45%' }}>
        <div className={classes.container}>
          {this.props.loading
            ? <div className={classes.bg} ><WrapperFormattedMessage id='serviceManagement.serviceCard.stateIsUpdating' /></div>
            : ''
          }
          <div className={classes.top}>
            <h5 className={classes.h5} title={item.serviceName}>Service : {item.serviceName}</h5>
            <div style={{ whiteSpace: 'nowrap' }}>
              {
                status !== 'Undefined' ?
                  <button
                    type='button'
                    disabled={!['inited', 'stopped'].includes(item.status)}
                    onClick={this.toggleDialog('destroy')}
                    className={'pt-button pt-minimal pt-icon-delete'}
                    title={intl.formatMessage({ id: 'destroy' })}
                  /> :
                  <button
                    type='button'
                    onClick={this.toggleDialog('add')}
                    className={'pt-button pt-minimal pt-icon-add'}
                    title={intl.formatMessage({ id: 'serviceManagement.serviceCard.initService' })}
                  />
              }
              {
                status === 'Idle' ?
                  <button
                    type='button'
                    disabled={!['inited', 'stopped', 'feeded'].includes(item.status)}
                    onClick={this.toggleDialog('start')}
                    className={'pt-button pt-minimal pt-icon-play'}
                    title={intl.formatMessage({ id: 'serviceManagement.serviceCard.start' })}
                  /> :
                  <button
                    type='button'
                    disabled={!['started', 'feeding'].includes(item.status)}
                    onClick={this.toggleDialog('stop')}
                    className={'pt-button pt-minimal pt-icon-pause'}
                    title={stopOrLoading}
                  />
              }
              <button
                type='button'
                onClick={this.toggleDialog('edit')}
                disabled={status !== 'Idle'}
                className={'pt-button pt-minimal pt-icon-edit'}
                title={intl.formatMessage({ id: 'serviceManagement.serviceCard.edit' })}
              />
              <button
                type='button'
                // disabled={status !== 'In Progress'}
                onClick={this.toggleDialog('reset')}
                className={'pt-button pt-minimal pt-icon-history'}
                title={intl.formatMessage({ id: 'reset' })}
              />
              <EditDialog
                editOpen={this.state.editOpen}
                toggleDialog={this.toggleDialog('edit')}
                info={item}
              />
              <CreateServiceDialog
                isOpen={this.state.addOpen}
                toggleDialog={this.toggleDialog('add')}
                info={item}
              />
              <CommonDialog
                title={intl.formatMessage({ id: 'serviceManagement.serviceCard.isDelete' }, { id: item.id })}
                isOpen={this.state.deleteOpen}
                toggleDialog={this.toggleDialog('delete')}
                handleSubmit={this.handleDeleteDialog}
              />
              <CommonDialog
                title={intl.formatMessage({ id: 'serviceManagement.serviceCard.destroyService' })}
                isOpen={this.state.destroyOpen}
                toggleDialog={this.toggleDialog('destroy')}
                handleSubmit={this.handleSubmit('destroy')}
              />
              <CommonDialog
                title={intl.formatMessage({ id: 'serviceManagement.serviceCard.stopService' })}
                isOpen={this.state.stopOpen}
                toggleDialog={this.toggleDialog('stop')}
                handleSubmit={this.handleSubmit('stop')}
              />
              <CommonDialog
                title={intl.formatMessage({ id: 'serviceManagement.serviceCard.startService' })}
                isOpen={this.state.startOpen}
                toggleDialog={this.toggleDialog('start')}
                handleSubmit={this.handleSubmit('start')}
              />
              <CommonDialog
                title={intl.formatMessage({ id: 'serviceManagement.queueList.resetData' })}
                isOpen={this.state.resetOpen}
                toggleDialog={this.toggleDialog('reset')}
                handleSubmit={this.handleSubmit('reset')}
              />
            </div>
          </div>
          <div className={classes.id}>
            <span className={classes.span}><WrapperFormattedMessage id='serviceManagement.serviceCard.deviceName' />: {item.deviceName}</span>
            <span className={classes.span}><WrapperFormattedMessage id='modelManagement.ModelDetail.createTime' />: {item.createTime}</span>
          </div>
          <div className={classes.progress} style={{ backgroundColor: color }}>
            {status}{status === 'In Progress' ? `: ${item.progress || '0%'}` : ''}
          </div>
          <div className={classes.detail}>
            {new TextField(`${intl.formatMessage({ id: 'processId' })}：`, item.processName).getRender()}
            {new TextField(`${intl.formatMessage({ id: 'productId' })}：`, item.productName).getRender()}
            {new TextField(`${intl.formatMessage({ id: 'serviceManagement.queueList.glassId' })}：`, item.glassName).getRender()}
            {new TextField(`${intl.formatMessage({ id: 'serviceManagement.dialogFilter.model' })}：`, item.modelName).getRender()}
          </div>
        </div>
      </div>
    )
  }
}

ServiceCard.propTypes = {
  item: PropTypes.object.isRequired,
  loading: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
  intl: PropTypes.object.isRequired,
}

export default connect(
  ({ auth, service, intl: { locale } }) => ({
    locale,
    token: auth.token,
    serviceList: service.serviceList,
  }))(injectIntl(ServiceCard))
