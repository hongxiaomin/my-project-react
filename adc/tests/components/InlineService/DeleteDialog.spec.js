import React from 'react'
import { DeleteDialog } from 'routes/Inline/routes/Service/components/ServiceCard/DeleteDialog'

describe('(Component) DeleteDialog', () => {
  const id = 1
  const isOpen = false
  const toggleDialog = jest.fn()
  const dispatch = jest.fn()
  it('render normally', () => {
    const wrapper = shallow(
      <DeleteDialog
        id={id}
        isOpen={isOpen}
        toggleDialog={toggleDialog}
        dispatch={dispatch}
      />,
    )
    wrapper.find('Button').last().simulate('click')
    expect(wrapper).toMatchSnapshot()
  })
})
