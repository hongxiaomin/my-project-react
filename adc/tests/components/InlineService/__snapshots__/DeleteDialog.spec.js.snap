// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) DeleteDialog render normally 1`] = `
<div
  className="Dialog-container"
>
  <Blueprint.Dialog
    className="dialog"
    isOpen={false}
    onClose={[Function]}
  >
    <div
      className="header"
    >
      <h6
        className="headerTitle"
      >
        Confirmation
      </h6>
    </div>
    <div
      className="dialogBody"
    >
      <br />
      <p
        className="cn"
      >
        是否删除id为1的这条数据?
      </p>
      <div
        className="btnGroup"
      >
        <Blueprint.Button
          className="btnCancel"
          onClick={[Function]}
        >
          Cancel
        </Blueprint.Button>
        <Blueprint.Button
          className="pt-button pt-intent-primary btnEdit"
          onClick={[Function]}
        >
          Save
        </Blueprint.Button>
      </div>
    </div>
  </Blueprint.Dialog>
</div>
`;
